.github/workflows/build_and_test.yaml

```diff
@@ -52,16 +50,33 @@ jobs:
         files: ./coverage.xml
         name: codecov-envoy-gateway
         verbose: true
+  
+  conformance-tests:
+    runs-on: ubuntu-latest
```

Comment:
we ideally want unit-tests, conformance and push to be on the same runner

Reply:
AFAIK there's no way to run steps within a job in parallel, hence splitting conformance out into its own matrix job so it's not too slow. We can use upload/download artifact steps to share the binaries between jobs (this is the gist of https://github.com/envoyproxy/gateway/issues/683). Are there other reasons for wanting everything on the same runner?

Reply:
the only one is to make sure the same binary is used between all steps, so recommend
* running the jobs sequentially in this PR and then
* #683 can be used to enhance the CI to be able to share builds across runners